### PR TITLE
Migrate LoggingSystemTests to JUnit 5

### LINKED ISSUE #17097: Migrate JavaLoggingSystemTests and Log4J2LoggingSystemTests to JUnit 5
Currently ignoring from the checkstyle rule since they need to be migrated

### PR DESCRIPTION
**Pull request by L00kian**
Closes #17097

### CODE REVIEW

File: spring-boot-project/spring-boot-tools/spring-boot-test-support/src/main/java/org/springframework/boot/testsupport/system/LocaleExtension.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+/**
+ * Internal JUnit 5 {@code @Extension} to override system locale for test execution.
+ *
+ * @author Ilya Lukyanovich
+ */
+public class LocaleExtension implements BeforeEachCallback, AfterEachCallback {
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-tools/spring-boot-test-support/src/main/java/org/springframework/boot/testsupport/system/LocaleExtension.java]: As this is only used in one place (`JavaLoggingSystemTests`), I'd prefer to manage it in `@BeforeEach` and `@AfterEach` methods in the class rather the pulling out an extension in a separate module.

File: spring-boot-project/spring-boot/src/test/java/org/springframework/boot/logging/AbstractLoggingSystemTests.java [hunk 2]
```diff
@@ -16,42 +16,39 @@
...
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.io.TempDir;
 
 import org.springframework.util.StringUtils;
 
 /**
- * Base for {@link LoggingSystem} tests.
+ * Base for {@link LoggingSystem} junit 5 tests.
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/test/java/org/springframework/boot/logging/AbstractLoggingSystemTests.java]: This line can be reverted now that the JUnit 4/5 split is gone.

File: spring-boot-project/spring-boot-tools/spring-boot-test-support/src/main/java/org/springframework/boot/testsupport/system/LocaleExtension.java (hunk 1 above)
Comment (wilkinsona) [reply spring-boot-project/spring-boot-tools/spring-boot-test-support/src/main/java/org/springframework/boot/testsupport/system/LocaleExtension.java]: This was marked as resolved but it doesn't seem to have been. Do you disagree about there being no need for the extension, @L00kian?

**Comment by L00kian** [comment]
@wilkinsona Yep, sorry I was absolutely certain that I did delete the class and its usage. Done now

**Comment by wilkinsona** [comment]
Thanks very much, @L00kian.

**Pull request closed without merging**